NuGet/NuGet.Client
# Use OrdinalIgnoreCase for all Source and Package source mapping implementation

## Bug



Fixes: https://github.com/nuget/home/issues/11182
Fixes: https://github.com/NuGet/Home/issues/11241

Regression? Last working version:

## Description
@kartheekp-ms  noticed we have inconsistent use of string comparison approach, after discussion we decided to use OrdinalIgnoreCase only for "Source and package source mapping", after my current changes I don't see any breaking or big difference after this change.

nuget.config used for my manual testing:

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <clear />
    <add key="encyclopaedia" value="https://api.nuget.org/v3/index.json" />
    <add key="Encyclopaedia"  value="C:\Users\xxxx\1111" />
    <add key="encyclopædia" value="C:\Users\xxxx\2222" />
    <add key="Encyclopædia" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vssdk/nuget/v3/index.json" />
  </packageSources>
</configuration>
```

I didn't do changes in following places due to breaking/unwanted change:

- See comment  https://github.com/NuGet/NuGet.Client/pull/4264#discussion_r709646405 https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs#L419

-  See comment https://github.com/NuGet/NuGet.Client/pull/4264#discussion_r709649975
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Configuration/Settings/Items/SourceItem.cs#L80

In Apex tests I moved create nuget.config part before solution create action.

## PR Checklist


- **Tests**
  - **OR**
  
  - **OR**

- **Documentation**
  
  - **OR**
